openupgrade_scripts/scripts/hr/16.0.1.1/post-migration.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2023 Coop IT Easy (https://coopiteasy.be)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from openupgradelib import openupgrade
+
+_logger = logging.getLogger(__name__)
+
+
+def warn_about_dataloss(cr, source_relation_table, relation_comodel_field):
+    """Warn user about data loss when migrating data from many2many to
+    many2one.
+
+    :param source_relation_table: The many2many relation table
+    of the model that will on the 'one' side of the relation
+    :param relation_comodel_field: The name of the column containing ids
+    of the 'many' part of the new relation.
+    """
+    openupgrade.logged_query(
+        cr,
+        """
+        SELECT DISTINCT %(relation_comodel_field)s
+        FROM %(source_relation_table)s
+        WHERE %(relation_comodel_field)s IN (
+            SELECT %(relation_comodel_field)s
+            FROM %(source_relation_table)s
+            GROUP BY %(relation_comodel_field)s
+            HAVING COUNT(*) > 1
+        )
+        """
+        % {
+            "source_relation_table": source_relation_table,
+            "relation_comodel_field": relation_comodel_field,
+        },
+    )
+    for res in cr.fetchall():
+        _logger.error(
+            "hr.plan.activity.type(%s,) is linked to several hr.plan. "
+            "hr.plan.activity.type can only be linked to one hr.plan. "
+            "Fix these data before migrating to avoid data loss.",
+            res[0],
+        )
+
+
+def m2m_to_o2m(
```

Comment:
I wonder if such a function is generic enough to go into `openupgradelib` ?

Reply:
Well, it's not a usual case, but if it can be parametrized, then a good host is `openupgradelib`.